From 04d0a15363cfb2443ab290b9c0d102ccb3b6e02b Mon Sep 17 00:00:00 2001
From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Thu, 14 Jun 2012 13:19:18 +0200
Subject: [PATCH 16/19] ARM i.MX6 clk: initialize some video clocks

Add clock lookups for IPU1/2 and their DI0/1 and for the
HDMI TX. Set the video PLL to 650 MHz and source the
video clocks from it.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 arch/arm/mach-imx/clk-imx6q.c |   19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index 4233d9e..049a988 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -439,6 +439,25 @@ int __init mx6q_clocks_init(void)
 	clk_register_clkdev(clk[cko1_sel], "cko1_sel", NULL);
 	clk_register_clkdev(clk[ahb], "ahb", NULL);
 	clk_register_clkdev(clk[cko1], "cko1", NULL);
+	clk_register_clkdev(clk[ipu1], "bus", "2400000.ipu");
+	clk_register_clkdev(clk[ipu1_di0], "di0", "2400000.ipu");
+	clk_register_clkdev(clk[ipu1_di1], "di1", "2400000.ipu");
+	clk_register_clkdev(clk[ipu2], "bus", "2800000.ipu");
+	clk_register_clkdev(clk[ipu2_di0], "di0", "2800000.ipu");
+	clk_register_clkdev(clk[ipu2_di1], "di1", "2800000.ipu");
+	clk_register_clkdev(clk[hdmi_isfr], "isfr", "120000.hdmi");
+	clk_register_clkdev(clk[hdmi_iahb], "iahb", "120000.hdmi");
+
+	clk_set_rate(clk[pll5_video], 650000000);
+	clk_set_parent(clk[ipu1_di0_sel], clk[ipu1_di0_pre]);
+	clk_set_parent(clk[ipu1_di1_sel], clk[ipu1_di1_pre]);
+	clk_set_parent(clk[ipu2_di0_sel], clk[ipu2_di0_pre]);
+	clk_set_parent(clk[ipu2_di1_sel], clk[ipu2_di1_pre]);
+
+	clk_set_parent(clk[ipu1_di0_pre_sel], clk[pll5_video]);
+	clk_set_parent(clk[ipu1_di1_pre_sel], clk[pll5_video]);
+	clk_set_parent(clk[ipu2_di0_pre_sel], clk[pll5_video]);
+	clk_set_parent(clk[ipu2_di1_pre_sel], clk[pll5_video]);
 
 	for (i = 0; i < ARRAY_SIZE(clks_init_on); i++)
 		clk_prepare_enable(clk[clks_init_on[i]]);
-- 
1.7.10.4

